### DISCUSSION TITLE
AGW Upgrade 1.3.3 -> 1.4 Failing

### DISCUSSION THREAD

**Discussion by jpad-freedomfi** [magma/magma#5643] [Show and tell]
Hi All, 

Sharing experience of our first upgrade attempt from 1.3.3 to 1.4 on AGW. 

Our orc8r is 1.3.

**Network setup:**
- Bridged configuration (i.e. uplink_br0 for WAN)
- Static IP allocations for Subscribers
- Static IP configured for SGi Mgmt address (on uplink_br0)
- Multi-APN = true

**Procedure used for update:**

1. edit `/etc/apt/sources.list.d/packages_magma_etagecom_io.list` to point to 1.4.0 repo
2. apt update
3. apt upgrade magma -o Dpkg::Options::="--force-overwrite"

**Experience**

- First attempt to update, the update seemed to go okay, services were updated, networking remained functional. 
- looking at journalctl -f services were in a crash loop
- still debugging what was failing

Next I tried to disable stateless incase that was causing the issue

- disabled stateless in mme 
- disabled stateless in spgw
- disabled stateless in sessiond

Did not seem to effect the crash loop

Next I tried to roll back to 1.3.3

- Roll back seemed to work, UE regeistered and services looked happy. 

Next I attempted to upgrade to 1.4 again

- this time network connection was lost with AGW. 
- console connection showed no IP on uplink_br0 or eth0
- attempted to resurrect by deleting gateway.mconfig to allow it start fresh. 
- this lead to IP obtained via DHCP but it still was crash looping it seemed 

I'll update when we know more about what happened. 

Full terminal log attached and output of console after networking was broken. 

[lab1_agw_1.3.3_upgrade_attempt1_1.4_consoleoutput.txt](https://github.com/magma/magma/files/6177814/lab1_agw_1.3.3_upgrade_attempt1_1.4_consoleoutput.txt)
[lab1_agw_1.3.3_upgrade_attempt1_1.4.txt](https://github.com/magma/magma/files/6177815/lab1_agw_1.3.3_upgrade_attempt1_1.4.txt)

**Comment by pshelar** [comment]
One of issue is related setting def gw, I am surprised to see there is no default GW route configured. I guess all interfaces are in vlans.
anyways I have posted https://github.com/magma/magma/pull/5646 to handle it.

**Comment by jpad-freedomfi** [comment]
Hi @pshelar How did we get to be without a default route?

We have static SGI mgmt IP set. Here is the relavent portion of gateway json config. This was same config used with 1.3.3

```
"root":{8 items
"cellular":{3 items
"dns":{3 items
"dhcp_server_enabled":false
"enable_caching":false
"local_ttl":0
}
"epc":{4 items
"ip_block":"192.168.128.0/24"
"nat_enabled":false
"sgi_management_iface_gw":"10.200.82.1"
"sgi_management_iface_static_ip":"10.200.82.42/24"
}
"ran":{2 items
"pci":260
"transmit_enabled":true
}
}
```

**Reply by pshelar** [reply]
AGW reads the config from linux routing table, can you check output of `ip r sh`
there is a change in default GW in 1.4. thats why you dont see the issue in 1.3.
